lib/plugins/generic_place.js

```diff
@@ -37,7 +38,7 @@ function inject (bot, { version }) {
       await bot.lookAt(referenceBlock.position.offset(dx, dy, dz), options.forceLook)
     }
     // TODO: tell the server that we are sneaking while doing this
-    const pos = referenceBlock.position
+    const pos = referenceBlock.position.plus(faceVector)
```

Comment:
This is probably what broke 1.8 sign tests. You should not send `.plus(faceVector)` on the packet, and the methods that use _genericPlace already do this, so no need to return it either
https://github.com/PrismarineJS/mineflayer/blob/ec683de1f714db593c4843bddf8e45595eb637ea/lib/plugins/place_entity.js#L48
https://github.com/PrismarineJS/mineflayer/blob/ec683de1f714db593c4843bddf8e45595eb637ea/lib/plugins/place_block.js#L10

Reply:
The reference block is not the position you should send in the packet, the packet position needs to the position you're _placing the block in_. If you use the reference block position here you end up trying to put the block _into_ an existing block. The furnace test bug is linked to this problem.

Those `dest` calculations were added by me for the purpose of fixing race conditions with relying on _generic_place to do the calculation, but the API pre-dates this PR and I didn't want to break it.

Reply:
You're correct about this btw, thanks